stdlib/Equals.dora

```diff
@@ -0,0 +1,7 @@
+trait Equals {
+  fun equals(other: Self) -> Bool;
+}
+
+impl Equals for Int {
+  internal fun equals(other: Int) -> Bool;
```

Comment:
I think we should have a rule that a trait can be directly implemented by a class if it defines the function the trait requires. This would turn

```impl Equals for Int```

into a valid implementation.